src/arch/mod.rs

```diff
@@ -0,0 +1,9 @@
+#[cfg(target_arch = "riscv64")]
+mod riscv64;
+
+pub fn new_arch() -> impl Architecture {
+    #[cfg(target_arch = "riscv64")]
+    return riscv64::Riscv64::new();
+}
```

Comment:
So if the target arch isn't specified it simply doesn't compile :D That's good. I think it would be nice to add an `unreachable!()` at the end to explain to the user what went wrong? Something like `unreachable!("No target specified, run gen.sh")`? If that's necessary?

Reply:
Good idea